torch/_dynamo/convert_frame.py

```diff
@@ -158,6 +158,7 @@ def _fn(*args, **kwargs):
         torch_rng_state = torch.random.get_rng_state()
         if torch.cuda.is_available():
             cuda_rng_state = torch.cuda.get_rng_state()
+        allow_tf32 = torch._C._get_cublas_allow_tf32()
```

Comment:
Is global state CUDA-specific? Otherwise I would consider using `float32_matmul_precision` which should also cover other potential backends (`_get_cublas_allow_tf32` just checks this setting behind the scenes anyway).